### PR TITLE
Use http.target to drop non application spans

### LINKED ISSUE #26776: OpenTelemetry: High span name cardinality for invalid HTTP request spans
### Describe the bug

This is about the remaining issue from #26216:
When using the `quarkus-opentelemetry` extension and observing Vert.x HTTP server request spans, the span names for **invalid HTTP requests** (resulting in status 400 or also 501 responses) contain the full request URI path. The [OTEL spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name) declares such high cardinality span names as unsuitable. 

In practice, this means the span selection combobox in the Jaeger UI gets fill with all kinds of weird URI paths of bot requests trying to exploit some vulnerability.
```
/someWeirdPathTryingToExploitSomeVulnerability?someParamX
```

_Some_ of these span names still can get renamed by using a "match-all" Vert.x route with a failure handler (invoking `ctx.request().routed(name)` there), 

But there are kinds of invalid HTTP requests, for which no route failure handler will get invoked. This happens for example with HTTP requests with an invalid `Content-Length` header value.

### Expected behavior

I would expect some fixed span name (e.g. the HTTP request method name) to be used as span name for such invalid HTTP requests - **if** no request route name got set/applied (i.e. no named route got matched and `ctx.request().routed(name)` wasn't invoked).

### Actual behavior

High cardinality span names are used.

### How to Reproduce?

Make invalid HTTP requests to a quarkus-opentelemetry application having a Vert.x HTTP server endpoint and observe created span names.
Examples:
`printf 'GET invalid__no_leading_slash HTTP/1.0\r\n\r\n' | nc -q 3 localhost 80`
`printf 'GET /something%%%%3 HTTP/1.0\r\n\r\n' | nc -q 3 localhost 80`

`printf 'GET /something HTTP/1.1\r\nContent-Length: 18446744073709551615\r\n\r\n' | nc -q 3 localhost 80`
=> no Vert.x route getting invoked

`printf 'GET /something HTTP/5.0\r\n\r\n' | nc -q 3 localhost 80`
=> this actually returns status 501; no Vert.x route getting invoked

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

openjdk 17.0.3 2022-04-19

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

2.10.2

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by radcortez**
- Fixes #26776

### CODE REVIEW

**Comment by gsmet** [comment]
@kdubb could you have a look at this one?

**Comment by brunobat** [comment]
@radcortez can you please add a test?

**Comment by radcortez** [comment]
> @radcortez can you please add a test?

**Comment by brunobat** [comment]
> > @radcortez can you please add a test?
> 
> Tests already exist:
> 
> * io.quarkus.opentelemetry.deployment.NonAppEndpointsDisabledTest
> * io.quarkus.opentelemetry.deployment.NonAppEndpointsDisabledWithRootPathTest
> * io.quarkus.opentelemetry.deployment.NonAppEndpointsEnabledTest
> * io.quarkus.opentelemetry.deployment.NonAppEndpointsEqualRootPath
> 
> This changes the underlying implementation to use the `http.target` to match the `uri` instead of the name, because the name may not always be the `uri`

It's just because this behaviour seems not to be considered in the current tests... Something that will break if `http.target` usage is reverted.

**Comment by radcortez** [comment]
> It's just because this behaviour seems not to be considered in the current tests... Something that will break if `http.target` usage is reverted.

What do you mean?

The old implementation used the span name to drop the non-application uris (like health or the dev console). This was incorrect because it assumes that the span name for http is always the uri (and we were setting that up previously). Due to the high cardinality issues in #26776, #26216, and #26694, it is better to remove the route resolution for uri, which was setting the name, and just keep the OTel way to resolve the name (and update the route if available).

This means that the span name is not a reliable way to drop endpoints, so that is why we are changing the implementation to use `http.target`. All the relevant tests pass, so our new behavior is still the same. What other tests would you like to see?

**Review by brunobat — APPROVED** [review]
Explanation is good. Thanks @radcortez

File: extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/DropTargetsSampler.java [hunk 1]
```diff
@@ -8,25 +8,26 @@
...
     public SamplingResult shouldSample(Context parentContext, String traceId, String name, SpanKind spanKind,
             Attributes attributes, List<LinkData> parentLinks) {
 
         if (spanKind.equals(SpanKind.SERVER)) {
-            for (String dropName : dropNames) {
-                if (name.startsWith(dropName)) {
-                    return SamplingResult.drop();
-                }
+            String target = attributes.get(SemanticAttributes.HTTP_TARGET);
+            // TODO - radcortez - Match /* endpoints
```
Comment (gsmet) [inline extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/DropTargetsSampler.java:28]: @radcortez hmmm I just saw this TODO. Is it something that needs fixing soon?

Comment (radcortez) [reply extensions/opentelemetry/opentelemetry/runtime/src/main/java/io/quarkus/opentelemetry/runtime/tracing/DropTargetsSampler.java:28]: Not urgent, the previous implementation was working this way.

**Pull request merged**